src/js/extends.js

```diff
@@ -49,6 +51,11 @@ const extendsFn = function(superClass, subClassMethods={}) {
   let methods = {};
 
   if (typeof subClassMethods === 'object') {
+    if (typeof subClassMethods.init === 'function') {
+      log.warn('Constructor logic via init() is deprecated; please use constructor() instead.');
+      subClassMethods.constructor = subClassMethods.init;
+      delete subClassMethods.init;
```

Comment:
I'm not sure on this one either, but we may want to consider leaving it in case older external code points to the init directly for some reason. Probably better they get the warning than a broken function.